### ISSUE TITLE
Photo-Realistic Single Image Super-Resolution Using a Generative Adversarial Network

### ISSUE THREAD

**Issue by reyllama** [reyllama/paper-reviews#29]
![image](https://user-images.githubusercontent.com/46308407/105038549-d0402a80-5aa2-11eb-820b-402dabd93487.png)

# Key Idea

- Incorporate GAN architecture to Image Super-Resolution Task.
- Use residual blocks with skip connections to train deep convolutional model.
- Employ perceptual loss (L2 distance between VGG feature maps) instead of pixel-wise L2 distance that typically blurs the output.

# Previous Approaches

### Image Super-Resolution

- Neighborhood Embedding
- Convolutional models aiming to minimize pixel-wise L2 reconstruction loss.

![image](https://user-images.githubusercontent.com/46308407/105039119-8441b580-5aa3-11eb-9dee-5e313fd553b7.png)

**Note** <br>
Previous models trying to minimize L2 loss yields high PSNR (Peak Signal to Noise Ratio) but perceivably unsatisfying result. <br>
Hence, the evaluation of SRGAN relies heavily on MOS, Mean Opinion Score rated by human raters.

### Convolutional Neural Networks

- Batch Normalization
- Residual Blocks
- Learned Upscaling Filters (Pixel Shuffle) 

# Formulation

### Architecture

![image](https://user-images.githubusercontent.com/46308407/105039303-bbb06200-5aa3-11eb-83ca-ffe133bb075a.png)

- Prepare HR and LR image pair by applying Gaussian filter(blurs) and down-sample(bi-cubic) from original HR image.
- Use residual blocks with skip connection, and Parametric ReLU as activation. (Generator)
- Discriminator uses Leaky ReLU activation as proposed in DCGAN paper.

### Loss Function

![image](https://user-images.githubusercontent.com/46308407/105039590-15189100-5aa4-11eb-9ea7-5d8177bbf9fb.png)

![image](https://user-images.githubusercontent.com/46308407/105039641-2366ad00-5aa4-11eb-94c0-68db755df18d.png)

VGG loss for content loss. (High-level feature distance)

![image](https://user-images.githubusercontent.com/46308407/105039693-35485000-5aa4-11eb-8e18-df8ccd45a5ea.png)

Modified Vanilla GAN loss. (Stronger against vanishing gradient)

# Experiments / Results

- Pretrain SRResNet (Generator) with MSE loss for 10e6 iterations.
- 10e5 with lr 1e-4, and 10e5 with lr 1e-5 for whole SRGAN.
- 26 raters evaluated images from score 1 to 5 (MOS)

![image](https://user-images.githubusercontent.com/46308407/105039971-98d27d80-5aa4-11eb-879e-4a01a01b147e.png)

- Note quantitative results are not that favorable. They rely on MOS, claiming that existing metric doesn't reflect human perception quality well.

![image](https://user-images.githubusercontent.com/46308407/105040044-b30c5b80-5aa4-11eb-92c3-0adfbbc27525.png)

![image](https://user-images.githubusercontent.com/46308407/105040095-c7505880-5aa4-11eb-9074-e7f85d821fed.png)

- In terms of MOS, SRGAN performs well.

![image](https://user-images.githubusercontent.com/46308407/105040226-ec44cb80-5aa4-11eb-872c-75dd0646e54d.png)

- Skip Connections are central for training deep models.

![image](https://user-images.githubusercontent.com/46308407/105040303-054d7c80-5aa5-11eb-8967-872235b8ebca.png)

- Some more visual outputs. These are the (only) indicators of SRGAN's performance.

# Comments

- 전반적으로 아주 쉬운 논문. 구조도 단순하고 아이디어도 모두 직관적.
- Perceptual Loss도 지금에 와서는 굉장히 보편화된 개념이라서, 2017년 당시에 새로웠을 아이디어. (Neural Style Transfer도 2017년)
- 눈으로 보기에 좋을 결과들이 나온 건 맞는데, 우선 논문에서 그렇게 많은 샘플을 제공하고 있지도 않고, quantitative outcome이 꽤 별로라서, 성과에 대해서는 조금 애매한 부분은 있는듯 하다.
- PSNR, SSIM은 내가 생각해도 좋은 metric은 아닌듯하다. 이 부분에 대한 설명이 더 있었으면 좋았을 듯. 예컨대 PSNR은 높게 나오는데 visual quality는 굉장히 떨어지는 예시(일종의 반례)를 제안한다던지. FID를 비판하고 Precision/Recall을 제안할 때 그랬던 것처럼.
- 지금의 SR SOTA가 궁금. 사실 4x upsampling이면 그렇~게 어려운 테스크는 아닐 것 같아서.
